projects/cds/src/merchandising/facade/cds-merchandising-user-context.service.ts

```diff
@@ -33,9 +33,11 @@ export class CdsMerchandisingUserContextService {
       map(([productId, category, facets]: [string, string, string]) => {
         const userContextData: MerchandisingUserContext = {
           category,
-          productId,
           facets,
         };
+        if (productId) {
```

Comment:
from my understanding, you do not want to include the it to the userContextData if no productId exists? That is why you pulled it out above right?

If above is not the case, then you can do this

const .. : xxx = {
category,
products/productCodes: [productId],
facets
}

Reply:
Yea, I pulled it out to avoid cases of `products: [undefined]` or `products: undefined`